server/tasks/process_file.py

```diff
@@ -223,61 +139,26 @@ async def async_process_streaming_output(
             retries -= 1
             time.sleep(1)
 
-    input_job_running = True
-
-    data = await consumer.getmany(timeout_ms=timeout_ms, max_records=batch_size)
-
-    while input_job_running:
-        for tp, messages in data.items():
-            if messages:
-                logger.debug(f"Handling {messages=} in topic {tp.topic}")
-                data = [msg.value for msg in messages]
-                result_handler(data)
-                logger.debug(f"Handled {len(messages)} messages in topic {tp.topic}")
-            else:
-                logger.debug(f"No messages in topic {tp.topic}")
-
-        if not data:
-            logger.info(f"No messages in any topic")
-
-        job = process_file_streaming.AsyncResult(input_job_id)
-        # we are getting packets from the output topic here to check if its empty and continue processing if its not
-        data = await consumer.getmany(timeout_ms=timeout_ms, max_records=batch_size)
-        # TODO no way to recover here if connection to main app is lost, job will be stuck at "PENDING" so this will loop forever
-        if job.status in ["SUCCESS", "FAILURE", "REVOKED"] and len(data.items()) == 0:
-            input_job_running = False
-            logger.info(f"Input job done, stopping output job")
-        else:
-            logger.info(f"Input job still running, keeping output job running")
-
-    await consumer.stop()
-
-
-@app.task(
-    name="process_streaming_output",
-    track_started=True,
-    bind=True,
-    serializer="pickle",
-    task_time_limit=settings.task_time_limit_sec,
-)
-def process_streaming_output(
-    self,
-    input_job_id: str,
-    output_topic_name: str,
-    result_handler: ResultHandler,
-    batch_size: int,
-):
     try:
-        asyncio.run(
-            async_process_streaming_output(
-                input_job_id, output_topic_name, result_handler, batch_size
-            )
-        )
-    except Exception as e:
-        # Set own status to failure
-        self.update_state(state=states.FAILURE)
-
-        logger.error(msg=e)
-
-        # Ignore the task so no other state is recorded
-        raise Ignore()
+        while not input_done.is_set():
+            data = await consumer.getmany(timeout_ms=timeout_ms, max_records=batch_size)
+            for tp, messages in data.items():
```

Comment:
while we're changing this, mind just renaming `tp` to `topic`?

Reply:
done :)